### PR TITLE
Spinners/progress elements

### LINKED ISSUE #277: Spinner center has background that overlays poorly
In the image below (`Auth` block), the white background used to hide the inner part of the spinner's circle hides part of the underlying input field's border. This makes overlay spinners tricky to use.

screen shot 2018-01-10 at 4 46 28 pm

@TejasQ do you know a fix for it without changing how the spinner is made? Or shall I just go ahead and replace it with an SVG?

### PR DESCRIPTION
**Pull request by peterszerzo**
Fixes #277. Test like so:
`cd packages/showcase`
`yarn start`
`http://localhost:3000/components/loaders/

### CODE REVIEW

**Comment by TejasQ** [comment]
Fun fact, `cd packages/showcase` isn't necessary because the top-level `yarn start` runs `cd` for you!

**Comment by TejasQ** [comment]
I LOVE THAT THERE'S A PROGRESS BAR NOW!

File: packages/showcase/pages/components/loaders.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import * as React from "react"
+import { Card, CardHeader, Spinner, Progress } from "@operational/components"
+
+import Layout from "../../components/Layout"
+import Table from "../../components/PropsTable"
+import Playground from "../../components/Playground"
+
+const spinnerSnippet = `
+<Spinner/>
```
Comment (TejasQ) [inline packages/showcase/pages/components/loaders.js]: What props does this support?

Comment (peterszerzo) [reply packages/showcase/pages/components/loaders.js]: Just the standard `css` and `className` - I should make that clear for all components actually..

Comment (TejasQ) [reply packages/showcase/pages/components/loaders.js]: COOL I LOVE IT

File: packages/components/src/Progress.tsx [hunk 2]
```diff
@@ -18,65 +19,39 @@ const Container = glamorous.div(
...
-const BarContainer = glamorous.div(
-  {
-    width: "100%",
-    height: "100%",
-    overflow: "hidden"
-  },
-  ({ theme }: { theme: Theme }) => ({
-    backgroundColor: theme.colors.gray30,
-    border: `1px solid ${theme.colors.gray20}`
+  ({ theme, fadeParent }: { theme: Theme; fadeParent: boolean }) => ({
```
Comment (TejasQ) [inline packages/components/src/Progress.tsx]: what is fadeParent

File: packages/components/src/Spinner.tsx [hunk 3]
```diff
@@ -8,81 +8,64 @@ export interface IProps {
...
-  <Container
-    key={props.id}
-    css={props.css}
-    className={props.className}
-    color={props.color}
-    spinnerSize={props.size}
-    spinDuration={props.spinDuration}
-  />
+  <Container key={props.id} css={props.css} className={props.className}>
+    <Element />
```
Comment (TejasQ) [inline packages/components/src/Spinner.tsx]: Can we rename Element to something more descriptive? 😄

File: packages/showcase/components/Layout.js [hunk 4]
```diff
@@ -2,12 +2,15 @@ import glamorous, { ThemeProvider } from "glamorous"
...
 import { baseStylesheet, darken } from "@operational/utils"
-import { OperationalUI } from "@operational/components"
+import { OperationalUI, Progress } from "@operational/components"
+import Router from "next/router"
 
 import Header from "../components/Header"
 import Sidenavigation from "../components/Sidenavigation"
 import nextConfig from "../next.config"
 
+Router.onRouteChangeComplete = () => {}
```
Comment (TejasQ) [inline packages/showcase/components/Layout.js]: ?

File: packages/components/src/Spinner.tsx (hunk 3 above)
Comment (peterszerzo) [reply packages/components/src/Spinner.tsx]: Good call!

File: packages/showcase/components/Layout.js (hunk 4 above)
Comment (peterszerzo) [reply packages/showcase/components/Layout.js]: I'm removing it now, it was testing code. Thanks for the catch! :)

File: packages/components/src/Progress.tsx (hunk 2 above)
Comment (peterszerzo) [reply packages/components/src/Progress.tsx]: If you include `fadeParent`, it adds a near-opaque white cover over the direct parent to make it clear that the content is not accessible/dated/something new coming.

Comment (TejasQ) [reply packages/components/src/Progress.tsx]: Nice! Is that a `prop`?

Comment (peterszerzo) [reply packages/components/src/Progress.tsx]: Yes, it's documented in the Progress' propDescription - but I copied in my response right above to make it clearer.

File: packages/components/src/Spinner.tsx [hunk 5]
```diff
@@ -23,22 +23,20 @@ const size = 30
 
 const Container = glamorous.div(({ theme }: { theme: Theme }): {} => ({
   label: "spinner",
-  margin: "auto",
   width: size,
-  height: size,
-  position: "relative",
-  transform: "translateZ(0)"
+  height: size
```
Comment (TejasQ) [inline packages/components/src/Spinner.tsx]: What is `size`?

Comment (peterszerzo) [reply packages/components/src/Spinner.tsx]: There a `const size = 30` a few lines up.

**Pull request merged**